client/src/partials/cash/modals/transfer.modal.js

```diff
@@ -0,0 +1,113 @@
+angular.module('bhima.controllers')
+.controller('CashTransferModalController', CashTransferModalController);
+
+CashTransferModalController.$inject = [
+  '$uibModalInstance', '$q', 'SessionService', 'CurrencyService', 'VoucherService', 'CashboxService', 'AccountService', 'cashBox', 'uuid'
+];
+
+/**
+ * @module cash/modals/CashTransferModalController
+ *
+ * @description This controller is responsible transfering money between auxillary cash and a virement account
+*/
+
+function CashTransferModalController(ModalInstance, $q, sessionService, currencyService, voucherService, cashBoxService, accountService, cashBox, uuid) {
+  var vm = this; 
+
+  /** Attaching service to the scope **/
+  vm.cashBox = cashBox;
+  vm.currencyService = currencyService; 
+  vm.cashBoxService = cashBoxService;
+
+  /** check if the cashbox id is provided**/
+  vm.cashBoxIdMissed = !cashBox.id;
+
+  /**
+  * @function submit
+  * @param {boolean} invalid it is true if the form is valid
+  * @description for submitting a dialog content
+  **/
+  function submit (invalid){
+    if (invalid) { return; }
+    var voucher = processVoucher();
+    var voucher_item = processVoucherItem(voucher.uuid);
+    var record = { voucher : voucher, voucher_item : voucher_item };
+    voucherService.create(record)      
+    .then(function (res){
+      vm.submited = true; //FIX ME, I need a goo idea
+    })
+    .catch(function (err){
+      vm.fail = true; //FIX ME, I need a goo idea
```

Comment:
That is a good idea.  However, the code doesn't do any error handling - it just changes the UI to a generic message.  My proposal does the same thing.

If you want to have a specific error message (which is a good idea), you could also do this:
```js
.catch(function (error) {
  vm.httpError = error;
});
```
and conditionally show this error message when the server encounters and error.  This is a pattern used by @lomamech  quite a bit.